lib/recog/nizer.rb

```diff
@@ -41,6 +74,45 @@ def self.match(match_key, match_string)
     nil
   end
 
+  #
+  # Search all fingerprint dbs and attempt to find a matching
+  # {Fingerprint fingerprint}, stopping at the first hit. Returns `nil`
+  # when no matching database or fingerprint is found.
+  #
+  # @param match_string [String] Service banner to match
+  # @param [Hash] filters This hash contains filters used to limit the
+  #   results to just those from specific types of fingerprints.
+  #   The matched values are derived from the 'fingerprints' top level
+  #   element in the fingerprint DB XML or, in the case of 'protocol',
+  #   this value can be overridden at the individual fingerprint level.
+  # @option filters [String] :match_key Value from XML 'matches' or file name
+  # @option filters [String] :fp_type  Fingerprint type (service, os, etc.)
+  # @option filters [String] :protocol Protocol (ftp, smtp, etc.)
+  # @return (see Fingerprint#match)
+  def self.match_all_db(match_string, filters = {})
```

Comment:
This and the next method are nearly identical.  Is the only difference that one returns a single match or `nil` and the other returns an array of matches, possibly empty?  If so, just have this method call the other one and return `nil` if empty, the first otherwise.